x/incentives/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,235 @@
+package keeper_test
+
+import (
+	"time"
+
+	"github.com/c-osmosis/osmosis/x/incentives/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (suite *KeeperTestSuite) TestGRPCPotByID() {
+	suite.SetupTest()
+
+	// create a pot
+	potID, coins, startTime := suite.SetupNewPot(sdk.Coins{sdk.NewInt64Coin("stake", 10)})
+
+	// Ensure that a querying for a pot with an ID that doesn't exist returns an error
+	res, err := suite.app.IncentivesKeeper.PotByID(sdk.WrapSDKContext(suite.ctx), &types.PotByIDRequest{Id: 1000})
+	suite.Require().Error(err)
+	suite.Require().Equal(res, (*types.PotByIDResponse)(nil))
+
+	// Check that querying a pot with an ID that exists returns the pot.
+	res, err = suite.app.IncentivesKeeper.PotByID(sdk.WrapSDKContext(suite.ctx), &types.PotByIDRequest{Id: potID})
+	suite.Require().NoError(err)
+	suite.Require().NotEqual(res.Pot, nil)
+	suite.Require().Equal(res.Pot.Id, potID)
```

Comment:
Should we build an expected pot, and then a testing function to check that `actual pot == expected pot`? I think this would remove some of the code duplication

Reply:
It would be nicer